src/System.Text.Json/tests/Serialization/TestClasses.GenericCollections.cs

```diff
@@ -199,7 +199,7 @@ public class StringICollectionWrapper : ICollection<string>
 
         public int Count => _list.Count;
 
-        public bool IsReadOnly => ((ICollection<string>)_list).IsReadOnly;
+        public virtual bool IsReadOnly => ((ICollection<string>)_list).IsReadOnly;
```

Comment:
Presumably, these being virtual is necessary for the repro?

What happens if this wasn't virtual?

Reply:
Being virtual is not necessary for the repro. Alternative is for a new class to be added which implements ICollection<string> and returns `True` for `IsReadOnly()`.
It's the same construct, the key thing here is for the implementing-type under test to return `True` for `IsReadOnly()`.